### PR TITLE
prow: normalize when looking up / comparing github logins

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes https://github.com/kubernetes/test-infra/issues/4804

### CODE REVIEW

**Comment by BenTheElder** [comment]
This still needs testing and there are probably other places this needs fixing, but it's a start.

**Comment by xiangpengzhao** [comment]
Add some test cases for this?

**Comment by BenTheElder** [comment]
@xiangpengzhao yeah, there are probably other places that aren't fixed yet too. I just wanted to get it out before I went home, I will look at this more tomorrow.

**Comment by xiangpengzhao** [comment]
@BenTheElder Ah, I just notice "This still needs testing" in your first reply :)

**Comment by BenTheElder** [comment]
I'm pretty convinced there are other bugs like this and I make a great test case😂, I just wanted to get @cjwagner's eyes on what I had.

We might want to have some kind of generic login "normalization" method in case GitHub changes how `login`s work in the future, or it might just be fine to sprinkle in some `strings.ToLower`  like the PR does now. I meant to write a comment about that but left for dinner and then got distracted by the release shenanigans :-)

**Comment by cjwagner** [comment]
I like the idea of a generalizing the login normalization. We should define a `NormLogin` func in the `github` package: `var NormLogin = strings.ToLower`.

File: prow/github/client.go [hunk 1]
```diff
@@ -654,7 +654,7 @@ func (c *Client) UnassignIssue(org, repo string, number int, logins []string) er
 	}
```
Comment (cjwagner) [inline prow/github/client.go]: `assignee.Login` should be wrapped in the normalization func.

File: prow/github/client.go [hunk 2]
```diff
@@ -617,7 +617,7 @@ func (c *Client) AssignIssue(org, repo string, number int, logins []string) erro
...
```
Comment (cjwagner) [inline prow/github/client.go]: `assignee.Login` should be wrapped in the normalization func.

Comment (BenTheElder) [reply prow/github/client.go]: done.

**Comment by BenTheElder** [comment]
I added the `NormLogin` method and addressed the comments but this still needs some unit tests.

**Comment by BenTheElder** [comment]
Possibly `UnrequestReview` needs some `NormLogin` as well, will resume looking at this after meetings.

File: prow/github/types.go [hunk 3]
```diff
@@ -227,7 +230,7 @@ func (i Issue) IsPullRequest() bool {
 
 func (i Issue) HasLabel(labelToFind string) bool {
 	for _, label := range i.Labels {
-		if strings.ToLower(label.Name) == strings.ToLower(labelToFind) {
```
Comment (BenTheElder) [inline prow/github/types.go]: oops got a little too happy with NormLogin here :joy:

**Comment by BenTheElder** [comment]
This should be more or less good to go, possibly more tests. There are also likely other places (mungegithub? some plugins?) that should be using something like the `NormLogin` method but that can be done in followups.

File: prow/github/types_test.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	if !issue.IsAuthor(normalized) {
+		t.Errorf("expected issue.IsAuthor(%s) to be true", normalized)
+	}
+
+	if !issue.IsAssignee(capsTestLogin) {
+		t.Errorf("expected issue.IsAssignee(%s) to be true", capsTestLogin)
+	}
+	if !issue.IsAssignee(normalized) {
+		t.Errorf("expected issue.IsAssignee(%s) to be true", normalized)
+	}
```
Comment (cjwagner) [inline prow/github/types_test.go]: Can you add some test cases for casing besides the original and normalized? Those are the cases that are most interesting. For example `BENTHEELDER` and `BeNtHeElDeR` should both work for `IsAuthor` and `IsAssignee`

Comment (BenTheElder) [reply prow/github/types_test.go]: Done, I added a set of valid alternatives + an explicitly normalized entry.

**Comment by BenTheElder** [comment]
If we don't already have one that test definitely needs an issue, I've seen
this flake before (
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/4805/pull-test-infra-bazel/8177/
)

**Comment by cjwagner** [comment]
@BenTheElder Agreed I'm making an issue now.

**Pull request merged**